doc/application/application.rst

```diff
@@ -928,6 +928,10 @@ checking the following until one is found, in order:
 If :makevar:`DTC_OVERLAY_FILE` specifies multiple files, they will be merged in
 order.
 
+Additionally, the build system will automatically include the required overlay
+for MCUboot chain-loading (located in :file:`dts/comon/mcuboot.overlay`)
+whenever the :option:`CONFIG_BOOTLOADER_MCUBOOT` is set using Kconfig.
+
```

Comment:
An alternative solution would be to have 1 DTS source file for all Zephyr builds that included other DTS sources, just like we do with the 1 linker script that is shared between all ARM builds.

Something like this:

```
#include <device.dts>

#ifdef CONFIG_BOOTLOADER_MCUBOOT
#include <dts/comon/mcuboot.overlay>
#endif

```

Then we could get rid of the pollution to the core build system.

Reply:
I agree with this, will implement now